libraries/AP_Camera/AP_Camera_SoloGimbal.cpp

```diff
@@ -0,0 +1,89 @@
+#include "AP_Camera_SoloGimbal.h"
+#include <GCS_MAVLink/GCS.h>
+
+uint8_t AP_Camera_SoloGimbal::gopro_capture_mode;
+uint8_t AP_Camera_SoloGimbal::gopro_status;
+bool AP_Camera_SoloGimbal::gopro_is_recording;
+mavlink_channel_t AP_Camera_SoloGimbal::heartbeat_channel;
+
+// Toggle the shutter on the GoPro
+// This is so ArduPilot can toggle the shutter directly, either for mission/GCS commands, or when the
+// Solo's gimbal is installed on a vehicle other than a Solo.  The usual GoPro controls thorugh the 
+// Solo app and Solo controller do not use this, as it is done offboard on the companion computer.
+void AP_Camera_SoloGimbal::gopro_shutter_toggle()
+{
+    if (gopro_status != STATUS::GOPRO_CONNECTED) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "GoPro Not Available");
+        return;
+    }
+
+    const uint8_t gopro_shutter_start[4] = { 1, 0, 0, 0};
+    const uint8_t gopro_shutter_stop[4] = { 0, 0, 0, 0};
+
+    if (gopro_capture_mode == CAPTURE::MODE_PHOTO) {
+        // Trigger shutter start to take a photo
+        gcs().send_text(MAV_SEVERITY_INFO, "GoPro Photo Trigger");
```

Comment:
We tend not to tell people everything is going fine :-)

Similarly for other `send_text`s

Reply:
I prefer notifications that it actually DID work for things like this. Especially since you can't necessarily always tell from the live feed.